pandas/tests/test_nanops.py

```diff
@@ -355,22 +310,31 @@ def test_nanmean_overflow(self):
             assert result == np_result
             assert result.dtype == np.float64
 
-    def test_returned_dtype(self):
-
-        dtypes = [np.int16, np.int32, np.int64, np.float32, np.float64]
-        if hasattr(np, "float128"):
-            dtypes.append(np.float128)
+    @pytest.mark.parametrize(
+        "dtype",
+        [
+            np.int16,
+            np.int32,
+            np.int64,
+            np.float32,
+            np.float64,
+            getattr(np, "float128", None),
```

Comment:
Is there a good way to convert this to a skip if not available?

Reply:
I guess we could make this:

```
pytest.param(getattr(np, "float128", None), marks=pytest.skipif(not hasattr(np, "float128"))
```

Tough call whether the explicit skip is worth the added verbosity